isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -869,266 +857,356 @@ namespace Isis {
+        finalParameterValues.append(coefZ[i]);
+      }
+    }
+    if (nPointingCoefficients > 0) {
+      for (int i=0; i < nPointingCoefficients; i++) {
+        finalParameterValues.append(coefRA[i]);
+      }
+      for (int i=0; i < nPointingCoefficients; i++) {
+        finalParameterValues.append(coefDEC[i]);
+      }
+      for (int i=0; i < nPointingCoefficients; i++) {
+        finalParameterValues.append(coefTWI[i]);
+      }
+    }
 
-    if (!imageCSV) {
+  }
 
-      QString str("%1(t%2)");
 
-      if (nPositionCoefficients > 0) {
-        for (int i = 0; i < nPositionCoefficients; i++) {
-          finalParameterValues.push_back(coefX[i]);
-          if (i == 0)
-            parameterNamesList.append( str.arg("  X  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-        for (int i = 0; i < nPositionCoefficients; i++) {
-          finalParameterValues.push_back(coefY[i]);
-          if (i == 0)
-            parameterNamesList.append( str.arg("  Y  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-        for (int i = 0; i < nPositionCoefficients; i++) {
-          finalParameterValues.push_back(coefZ[i]);
-          if (i == 0)
-            parameterNamesList.append( str.arg("  Z  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-      }
-      if (nPointingCoefficients > 0) {
-        for (int i = 0; i < nPointingCoefficients; i++) {
-          finalParameterValues.push_back(coefRA[i] * RAD2DEG);
-          if (i == 0)
-            parameterNamesList.append( str.arg(" RA  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-        for (int i = 0; i < nPointingCoefficients; i++) {
-          finalParameterValues.push_back(coefDEC[i] * RAD2DEG);
-          if (i == 0)
-            parameterNamesList.append( str.arg("DEC  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-        for (int i = 0; i < nPointingCoefficients; i++) {
-          finalParameterValues.push_back(coefTWI[i] * RAD2DEG);
-          if (i == 0)
-            parameterNamesList.append( str.arg("TWI  ").arg("0") );
-          else
-            parameterNamesList.append( str.arg("     ").arg(i) );
-        }
-      }
+  /**
+   * @brief Takes in an open std::ofstream and writes out information which goes into the
+   * bundleout.txt file.
+   *
+   * @param fpOut The open std::ofstream object which is passed in from
+   * BundleSolutionInfo::outputText()
+   * @param errorPropagation Boolean indicating whether or not to attach more information
+   *     (corrections, sigmas, adjusted sigmas...) to the output.
+   */
+  void BundleObservation::bundleOutputString(std::ostream &fpOut, bool errorPropagation) {
 
-    }// end if(!imageCSV)
+    char buf[4096];
```

Comment:
Both of these outputs formats logic have consistent pieces between the two. Why the String output uses a character buffer and the CSV output uses a QString is odd and both should probably use one or the other.